### PR TITLE
Plugin outdate version

### PR DESCRIPTION
**Pull request by antgonza**
First review/merge https://github.com/biocore/qiita/pull/2662.

Notes (sending to qiita-admin for further discussion):
1. The deprecation flags are shown if the plugin is not active so this info is not stored in the DB. This means that during start if a plugin doesn't start correctly the artifacts will be marked as outdated until they get started properly. This can only happen during restart.
2. If an analysis was created with an old version of qiime2, it will be marked as deprecated - which is not 100% accurate but perhaps fine 
3. This doesn't affect the loading of artifacts/context in Redbiom but we will need to figure out there how to deal with deprecation and how often we will need to update things
4. For the artifact listing (how we create analysis) decided to not display deprecated artifacts as this will make the displaying nicer (less artifacts) but the other option is to display it and add something to show that it's deprecated. Note that you can still add the deprecated artifacts via the study page.

Some examples:
- How deprecated artifacts will look in an analysis:
deprecated-analysis

- Deprecated artifacts in study page:
deprecated-study

- Artifact listing now:
study-artifact-before
vs. before: 
study-artifact-now

### CODE REVIEW

File: qiita_pet/static/js/qiita.js [hunk 1]
```diff
@@ -314,17 +314,20 @@ function format_biom_rows(data, row, for_study_list = true, samples = null) {
...
+        var algorithm = info.algorithm;
+        if (!(algorithm in processing_method)) {
+          processing_method[algorithm] = {};
+        }
 
-      var data_type = info.data_type + ' (' + info.target_subfragment.join(', ') + ')';
-      if (!(data_type in processing_method[algorithm])) {
-        processing_method[algorithm][data_type] = [];
+        var data_type = info.data_type + ' (' + info.target_subfragment.join(', ') + ')';
+        if (!(data_type in processing_method[algorithm])) {
```
Comment (ElDeveloper) [inline qiita_pet/static/js/qiita.js]: Is this checking that data_type is a substring of processing_method.algorithm?

Comment (antgonza) [reply qiita_pet/static/js/qiita.js]: yup, basically this is grouping the artifacts per data_type and it doesn't exits, it creates it ... a classic:
```
myvar = {}
for x in y:
   if x not in myvar:
      myvar[x] = init
  .....
```

**Comment by antgonza** [comment]
Note that the changes are basically adding a new column to the qiita database, which solves the 2 issues @ElDeveloper raised. Also decided to add this new info into the admin summary and this is how a deprecated software looks:
screen shot 2018-09-12 at 8 23 12 am

File: qiita_pet/static/js/qiita.js (hunk 1 above)
Comment (ElDeveloper) [reply qiita_pet/static/js/qiita.js]: I see, thanks!

**Comment by ElDeveloper** [comment]
Thanks @antgonza, this looks great! Lets just wait for tests to finish.

**Pull request merged**